python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -354,70 +358,240 @@ def _(
+    )
+
+    # TODO: Support cloud storage
+    exists = Path(ir.path).exists()
+    if (
+        exists and executor_options.scheduler == "distributed"
+    ):  # pragma: no cover; Requires distributed testing
+        raise NotImplementedError(
+            "Writing to an existing path is not yet supported "
+            "by the distributed GPU streaming executor."
+        )
+    elif exists and ir.kind != "Parquet":  # pragma: no cover
+        # TODO: Test this.
         raise NotImplementedError(
-            "Writing to an existing path is not supported "
-            "by the GPU streaming executor."
+            "Writing to an existing path is not yet supported for {kind}."
         )
-    new_node = ir.reconstruct([child])
+
+    new_node = StreamingSink(
+        ir.schema,
+        ir.reconstruct([child]),
+        executor_options,
+        child,
+    )
     partition_info[new_node] = partition_info[child]
     return new_node, partition_info
 
 
-def _prepare_sink(path: str) -> None:
+def _prepare_sink_directory(path: str) -> None:
     """Prepare for a multi-partition sink."""
     # TODO: Support cloud storage
     Path(path).mkdir(parents=True)
 
 
-def _sink_partition(
+def _sink_partition_to_directory(
     schema: Schema,
     kind: str,
     path: str,
     options: dict[str, Any],
     df: DataFrame,
     ready: None,
 ) -> DataFrame:
-    """Sink a partition to disk."""
+    """Sink a partition to a new file."""
     return Sink.do_evaluate(schema, kind, path, options, df)
 
 
-@generate_ir_tasks.register(Sink)
-def _(
-    ir: Sink, partition_info: MutableMapping[IR, PartitionInfo]
+def _sink_partition_to_parquet_file(
+    schema: Schema,
+    path: str,
+    options: dict[str, Any],
+    finalize: bool,  # noqa: FBT001
+    writer: plc.io.parquet.ChunkedParquetWriter | None,
+    df: DataFrame,
+) -> plc.io.parquet.ChunkedParquetWriter | DataFrame:
+    """Sink a partition to an open Parquet file."""
+    # Set up a new chunked Parquet writer if necessary.
+    if writer is None:
+        metadata = plc.io.types.TableInputMetadata(df.table)
+        for i, name in enumerate(df.column_names):
+            metadata.column_metadata[i].set_name(name)
+
+        sink = plc.io.types.SinkInfo([path])
+        builder = plc.io.parquet.ChunkedParquetWriterOptions.builder(sink)
+        compression = options["compression"]
+        if compression != "Uncompressed":
+            builder.compression(
+                getattr(plc.io.types.CompressionType, compression.upper())
+            )
+
+        if options["data_page_size"] is not None:
+            builder.max_page_size_bytes(options["data_page_size"])
+        if options["row_group_size"] is not None:
+            builder.row_group_size_rows(options["row_group_size"])
+
+        writer_options = builder.metadata(metadata).build()
+        writer = plc.io.parquet.ChunkedParquetWriter.from_options(writer_options)
```

Comment:
@Matt711 - Perhaps there is a way we can define this logic in a shared utility that can be used both here and in https://github.com/rapidsai/cudf/pull/19015 ?

Reply:
Yeah I'll factor it out in #19015 once this PR gets merged